core/src/main/java/com/scalar/db/common/checker/OperationChecker.java

```diff
@@ -220,6 +227,40 @@ private void checkOrderings(Scan scan, TableMetadata metadata) {
     }
   }
 
+  private void checkOrderings(ScanAll scan, TableMetadata metadata) {
+    for (Scan.Ordering ordering : scan.getOrderings()) {
+      if (!metadata.getColumnNames().contains(ordering.getColumnName())) {
+        throw new IllegalArgumentException(
+            "The specified ordering column is not found. Invalid ordering: "
+                + ordering
+                + ", Operation: "
+                + scan);
+      }
+    }
+  }
+
+  private void checkConjunctions(ScanAll scan, TableMetadata metadata) {
+    for (Conjunction conjunction : scan.getConjunctions()) {
+      for (ConditionalExpression condition : conjunction.getConditions()) {
+        boolean isValid;
+        if (condition.getOperator() == Operator.IS_NULL
+            || condition.getOperator() == Operator.IS_NOT_NULL) {
+          // the value must be null if the operator is 'is null' or `is not null`
```

Comment:
Sorry, I don't get this. Why `ColumnChecker.requireNull` is used even for `IS_NOT_NULL`?

Reply:
It's just because we assume the conditional expression for `IS_NOT_NULL` is built with a null column, e.g., like [this](https://github.com/scalar-labs/scalardb/blob/e4ca22cd7acba6b0ba429a6f9c1ac27cfcdd339c/core/src/main/java/com/scalar/db/api/ConditionBuilder.java#L675-L677). I added this check referring to [this](https://github.com/scalar-labs/scalardb/blob/e4ca22cd7acba6b0ba429a6f9c1ac27cfcdd339c/core/src/main/java/com/scalar/db/common/checker/ConditionChecker.java#L69-L74). (But, I couldn't reuse itself since the check condition is a bit different.)

Reply:
Ah, thank. I had a wrong idea. I thought the check was for filtering actual scanned value, but any scan hasn't started yet here. This check is kind of pre-checking if the conditions are valid or not, and actual filtering is performed in the underlying relational database (e.g, SimpleSelectQuery)